### PR TITLE
OOR client 4/4: retry-backoff, outbox error handling, crash-resume tests

### PR DESCRIPTION
**Pull request by bhandras**
## Context

Final client PR in the OOR stack. Split 3 (#79) made the actor durable with TLV codecs and signing context. Split 3b (#119) added operator signature validation. This split adds **retry-backoff runtime behavior** so transient outbox failures don't drop sessions, and adds crash-resume test coverage for the co-sign window.

### Retry-backoff FSM state
`RetryBackoff` is introduced as an explicit outgoing FSM state that stores a resume snapshot, retry delay, and failure reason. Snapshot restore and resume logic can represent retry intent without losing session progress after a restart.

### Retry transitions and scheduling
Retry progression is driven through the durable actor boundary with explicit `RetryDueEvent` and `ScheduleRetryRequest` messages. `OutboxForState` emits the expected retry scheduling request for restored sessions so restart replay stays consistent with the normal runtime path.

### Outbox error handling
Retryable outbox failures are routed through retry-backoff states and durable events. Non-retryable errors terminate the session. This classification happens at the actor level so the FSM stays pure.

### Crash-resume and edge-case tests
- Actor resume test simulating a crash after server co-sign but before the client observes submit acceptance. Restores from persisted state, re-drives the outbox, and asserts stable package behavior through completion.
- Targeted unit tests for DriveEvent handling, late/terminal event behavior, and retryable vs non-retryable outbox error transitions.

## Storage model

This split extends split 3's replay behavior — it does not add a new durability backend.

| Concern | Storage/Boundary | Trigger |
|---|---|---|
| Retry scheduling intent | `ScheduleRetryRequest` outbox event | retryable outbox error |
| Crash/restart replay | durable mailbox + TLV checkpoint | actor restart |

The retry delay and resume snapshot are captured inside the `RetryBackoff` FSM state and persisted through the existing `OutgoingSnapshot` → checkpoint TLV path from split 3.

## Commits

1. **oor: add retry-backoff state for snapshot restore path**
2. **oor: wire retry-backoff transitions for outgoing snapshots**
3. **oor: make retry-backoff resumable**
4. **oor: add retry-backoff flow and outbox error handling**
5. **oor: add crash-resume test for co-sign**
6. **test: cover OOR actor/FSM edge cases**
7. **oor: reconcile split4 with split3 validation base**

## Not included

- Full transport RPC integration beyond typed envelopes.
- System-level restart/retry integration tests (unit/in-process focus here).

## Stack

Parent: #119 · Final split in the client OOR stack.

### CODE REVIEW

**Review by ellemouton — APPROVED** [review]

File: oor/transitions.go [hunk 1]
```diff
@@ -303,6 +442,53 @@ func (s *Failed) ProcessEvent(ctx context.Context, event Event,
...
+		return nil, fmt.Errorf("internal: missing session id")
+	}
+
+	snap, err := NewOutgoingSnapshot(env.SessionID, current)
+	if err != nil {
+		return nil, err
+	}
+
+	after := evt.RetryAfter
+	if after == 0 {
```
[the remark this answers is not in the document]
Comment (bhandras) [reply oor/transitions.go:503]: Addressed: switched to named constant (defaultRetryDelay) instead of inline literal.

**Comment by litbot-9000** [comment]
@ellemouton: review reminder
@bhandras, remember to re-request review from reviewers when ready

File: oor/actor_drive_event_retry_due_test.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
+package oor
```
Comment (ellemouton) [inline oor/actor_drive_event_retry_due_test.go]: btw - maybe tell the agent not to put all testst into their own file?

also - godocs for tests 🙏

File: oor/retry_backoff_test.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
+package oor
```
Comment (ellemouton) [inline oor/retry_backoff_test.go]: same here

**Review by ellemouton — APPROVED** [review]

File: oor/actor_drive_event_retry_due_test.go (hunk 2 above)
Comment (bhandras) [reply oor/actor_drive_event_retry_due_test.go]: Good call — cleaned this up.

I moved the retry-due drive-event roundtrip test into `actor_durable_message_test.go` (next to the other durable command roundtrip coverage) and added godoc comments for the test functions there.

File: oor/retry_backoff_test.go (hunk 3 above)
Comment (bhandras) [reply oor/retry_backoff_test.go]: Addressed this one as well.

I moved the retry-backoff tests out of `retry_backoff_test.go` into the existing `actor_drive_event_integration_test.go` coverage and added godoc comments on the moved tests.

File: oor/actor_drive_event_retry_due_test.go (hunk 2 above)
Comment (bhandras) [reply oor/actor_drive_event_retry_due_test.go]: done

File: oor/retry_backoff_test.go (hunk 3 above)
Comment (bhandras) [reply oor/retry_backoff_test.go]: done
